### PR TITLE
use Content-Disposition for downloading images

### LINKED ISSUE #2600: The "Download" button from saved shot details page is not correctly working (stage)
[Affected versions]:
- Firefox 54.0a2 (Build ID:20170407004017)
- Page Shot 6.0.10626

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisites]:
- The "xpinstall.signatures.required" is set to "false" and latest Firefox Screenshots (6.0.10626) custom build is installed on a new clean profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites and navigate to any website.
2. Click the Firefox Screenshots button from browser toolbar. 
3. Perform a selection and save it.
4. Click the "Download" button from top right corner of the page.
5. Observe the browser behavior.

[Expected result]:
- The "Download" window is opened in order to download the shot.

[Actual result]:
- The saved shot image is opened in the same tab.

[Notes]:
- Here is a screen recording with the issue:
download shot

### PR DESCRIPTION
**Pull request by dannycoates**
Annoyingly, the `filename` portion is non-trivial to encode but there's an existing module to handle it.

fixes #2600

### CODE REVIEW

File: server/src/pages/shot/view.js [hunk 1]
```diff
@@ -329,8 +330,8 @@ class Body extends React.Component {
           </div>
           <div className="more-shot-actions right">
             { trashOrFlagButton }
-            <a className="button secondary" href={ clipUrl } onClick={ this.onClickDownload.bind(this) }
-              title="Download the shot image" download={ clipFilename }>
+            <a className="button secondary" href={ clipDownloadUrl } onClick={ this.onClickDownload.bind(this) }
+              title="Download the shot image">
```
Comment (pdehaan) [inline server/src/pages/shot/view.js]: Does this need to be localized, or are we not doing server l10n yet?

Comment (dannycoates) [inline server/src/pages/shot/view.js]: it will. we haven't started server l10n yet

File: server/src/pages/shot/view.js [hunk 2]
```diff
@@ -298,12 +298,13 @@ class Body extends React.Component {
     }
 
     let clipUrl = null;
+    let clipDownloadUrl = null;
     if (clipNames.length) {
       let clipId = clipNames[0];
       let clip = this.props.shot.getClip(clipId);
       clipUrl = clip.image.url;
+      clipDownloadUrl = clipUrl + `?download=${encodeURIComponent(this.props.shot.filename)}`;
```
Comment (ianb) [inline server/src/pages/shot/view.js]: This value needs to be signed by the server, so someone can't do something like: `http://cdnlink/image/uuid?download=somethingevil.exe`

Comment (ianb) [inline server/src/pages/shot/view.js]: BTW, the /proxy code is an example of signing links

**Pull request merged**